_episodes/l3_03_test_fixture.md

```diff
@@ -0,0 +1,505 @@
+    void setBaseSalary(double employeeSalary); 
+    void setNumberYearsEmployed(double employeeNumberYearsEmployed); 
+    void SetBasicBonus(double employeeBonus);
+
+    void calcNetBonus(); //To calculate net bonus while considering experience.
+    void calcTaxAmount(); // To calculate tax to be paid based on salary with bonus
+    void calcNetSalary(); // To calculate net salary after adjusting tax and bonus.
+
+    // Getter functions.
+    string getName() const;
+    float getAge() const;
+    double getBasicSalary() const;
+    double getNumberYearsEmployed() const;
+    double getBasicBonus() const;
+    double getNetBonus() const;
+    double getTaxAmount() const;
+    double getNetSalary() const;
+
+    void displayInfo() const; 
+
+    // Destructor
+    ~Employee();
+};
+```
+
+For the definition part, we include only a few functions here. You can find the complete definition of this class in [employee.cpp](../code/Chapter3/employee.cpp).
+
+```cpp
+// Constructor definition
+Employee:: Employee(const string& employee_name, float employee_age, 
+                    double employeeSalary, double employeeNumberYearsEmployed,
+                    double employeeBonus)
+                    : age(employee_age), 
+                      base_salary(employeeSalary),
+                      number_years_employed(employeeNumberYearsEmployed),
+                      basic_bonus(employeeBonus),
+                      net_bonus(0),
+                      tax_amount(0),
+                      net_salary(0)
+{
+    setName(employee_name);
+    calcNetBonus();
+    calcTaxAmount();
+    calcNetSalary();
+
+}
+
+// Definition of function setName
+void Employee::setName(const string& employee_name) 
+{
+    if(employee_name == "")
+    {
+        throw invalid_argument("Name cannot be empty");
+    }
+    name = employee_name;
+}
+
+// Employee gets additional £1000 bonus if they have been employed for more than 10 years.
+void Employee::calcNetBonus() 
+{   
+    if(number_years_employed > years_exp_for_extra_bonus)
+    {
+        net_bonus = basic_bonus + extra_bonus;
+    }
+    else
+    {
+        net_bonus = basic_bonus;
+    }
+}
+
+// Function to calculate tex amount.
+void Employee::calcTaxAmount() 
+{
+    double salary_with_bonus = base_salary + net_bonus;
+    if(salary_with_bonus <= 10000)
+    {
+        tax_amount = 0.0;
+    }
+    else if(salary_with_bonus > 10000 && salary_with_bonus <= 20000)
+    {
+        tax_amount = 0.1*(salary_with_bonus-10000);
+    }
+    else if(salary_with_bonus > 20000 && salary_with_bonus <= 50000)
+    {
+        tax_amount = 0.1*10000 + 0.2*(salary_with_bonus-20000);
+    }
+    else if(salary_with_bonus > 50000)
+    {
+        tax_amount = 0.1*10000 + \
+                     0.2*30000 + \
+                     0.5*(salary_with_bonus-50000);
+    }
+}
+
+// Function to calculate the ney salary.
+void Employee::calcNetSalary() 
+{
+    net_salary = base_salary + net_bonus - tax_amount;
+}
+```
```

Comment:
A lot of code to shove in someone's face in a seminar. Maybe just briefly look over it live, include the really relevant parts in the text, and leave going over the actual code for the exercises?

You could even just include the function signatures and explain what the function does in plain English.

Reply:
Thanks for the suggestion. It has been resolved in 8d4ec5ce637e86c5d3f9ded79606867d056496a5